src/query/src/log_query/planner.rs

```diff
@@ -0,0 +1,353 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use catalog::table_source::DfTableSourceProvider;
+use datafusion::datasource::DefaultTableSource;
+use datafusion_common::ScalarValue;
+use datafusion_expr::utils::conjunction;
+use datafusion_expr::{col, lit, Expr, LogicalPlan, LogicalPlanBuilder};
+use datafusion_sql::TableReference;
+use datatypes::schema::Schema;
+use log_query::{ColumnFilters, LogQuery, TimeFilter};
+use snafu::{OptionExt, ResultExt};
+use table::table::adapter::DfTableProviderAdapter;
+
+use crate::log_query::error::{
+    CatalogSnafu, DataFusionPlanningSnafu, Result, TimeIndexNotFoundSnafu, UnimplementedSnafu,
+    UnknownTableSnafu,
+};
+
+const DEFAULT_LIMIT: usize = 1000;
```

Comment:
Isn't 1000 too much? 100 should be enough, shouldn't it?

Reply:
Looks like 1000 is preferred https://github.com/grafana/loki/issues/2226